architecture/0007-use-weblate-for-i18n.md

```diff
@@ -0,0 +1,27 @@
+# 7. Use weblate for i18n
+
+Date: 2020-05-07
+
+## Status
+
+Accepted
+
+## Context
+
+We have a request to provide a french translationed version of de-africa. We are using the i18next and i18next-react libraries with all our UI strings within en/translation.json. There are also two long life branches master & next representing our current production release (v7) and the next production release (v8). Due to the time lag between releases, there are hundreds more strings in the v7 translation.json versus the master version.
+
+It would not be workable or scalable for translators to provide these translations into the github repository directly as this requires a level of technical expertise and we won't have an easy way of keeping track of which strings haven't been translated yet. We also require some quality check mechanism to ensure the quality of the contributed translations.
+
+## Decision
+
+We will use hosted weblate at https://hosted.weblate.org/projects/terriajs/ to allow external translators to contribute to the project. There will be two projects, one connected to the master branch and another connected to the next branch.
```

Comment:
I'd like to see some extra details here about what we intend to do when the next branch becomes master. Should we:
* Try to port translations from v7 to v8?
* Provide instructions for contributors to do this?
* Deprecate/terminate the v7 translation project on some schedule without a way to port the translations?

Should we continue making the v7 weblate commit back to a `7.x.x` branch of terriajs and release it every `n` months? I don't have a good answer to any of these questions.

Reply:
That's a tricky one because currently there is a possibility of duplication if two different translators contribute translations for the same string but in different branches. I don't think that decision should block this ADR however i will add it as something to be addressed once we release v8.